lib/ops/opsservice/status.go

```diff
@@ -22,61 +22,101 @@ import (
-	log "github.com/sirupsen/logrus"
 )
 
-// CheckSiteStatus runs app status hook and updates site status appropriately.
+// CheckSiteStatus runs application status hook and updates cluster status appropriately
 func (o *Operator) CheckSiteStatus(key ops.SiteKey) error {
 	cluster, err := o.openSite(key)
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	if !cluster.app.Manifest.HasHook(schema.HookStatus) {
-		log.Debugf("%v does not have status hook", key)
+	// pause status checks while the cluster is undergoing an operation
+	switch cluster.backendSite.State {
+	case ops.SiteStateActive, ops.SiteStateDegraded:
+	default:
+		o.Infof("Status checks are paused, cluster is %v.",
+			cluster.backendSite.State)
 		return nil
 	}
 
-	ref, out, err := app.RunAppHook(context.TODO(), o.cfg.Apps, app.HookRunRequest{
-		Application: cluster.backendSite.App.Locator(),
-		Hook:        schema.HookStatus,
-		ServiceUser: cluster.serviceUser(),
-	})
-
-	if ref != nil {
-		defer func() {
-			err := o.cfg.Apps.DeleteAppHookJob(context.TODO(), *ref)
-			if err != nil {
-				log.Warningf("failed to delete hook %v: %v",
-					ref, trace.DebugReport(err))
-			}
-		}()
+	statusErr := cluster.checkPlanetStatus(context.TODO())
+	reason := storage.ReasonNodeDegraded
+	if statusErr == nil {
+		statusErr = cluster.checkStatusHook(context.TODO())
+		reason = storage.ReasonStatusCheckFailed
 	}
 
-	if err != nil {
-		req := ops.DeactivateSiteRequest{
+	if statusErr != nil {
+		err := o.DeactivateSite(ops.DeactivateSiteRequest{
 			AccountID:  key.AccountID,
 			SiteDomain: cluster.backendSite.Domain,
-			Reason:     storage.ReasonStatusCheckFailed,
-		}
-		if err := o.DeactivateSite(req); err != nil {
+			Reason:     reason,
+		})
+		if err != nil {
 			return trace.Wrap(err)
 		}
-		return trace.Wrap(err, string(out))
+		return trace.Wrap(statusErr)
 	}
 
-	if cluster.backendSite.State == ops.SiteStateDegraded && cluster.backendSite.Reason == storage.ReasonStatusCheckFailed {
-		req := ops.ActivateSiteRequest{
-			AccountID:  key.AccountID,
-			SiteDomain: cluster.backendSite.Domain,
+	// all status checks passed so if the cluster was previously disabled
+	// because of those checks, enable it back
+	if cluster.backendSite.State == ops.SiteStateDegraded {
+		if cluster.backendSite.Reason != storage.ReasonLicenseInvalid {
+			err := o.ActivateSite(ops.ActivateSiteRequest{
+				AccountID:  key.AccountID,
+				SiteDomain: cluster.backendSite.Domain,
+			})
+			if err != nil {
+				return trace.Wrap(err)
+			}
 		}
-		if err := o.ActivateSite(req); err != nil {
-			return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+// checkPlanetStatus checks the cluster health using planet agents
+func (s *site) checkPlanetStatus(ctx context.Context) error {
+	planetStatus, err := status.FromPlanetAgent(ctx, nil)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	for _, node := range planetStatus.Nodes {
+		if node.AdvertiseIP != "" && node.Status != status.NodeHealthy {
```

Comment:
That would be very surprising to find a node with en empty `AdvertiseIP` and consider it an `ok` outcome.